### PR TITLE
external : Add gmock for C++ testing

### PR DESCRIPTION
**Pull request by HONGCHAEHEE**
- This includes Google Mock and Google Test which is a unit testing library for the C++ programming language.
- CONFIG_GMOCK should be enabled.

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by pillip8282** [comment]
There are comflicts on the commit. please fix it first.

File: external/gmock/Kconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# For a description of the syntax of this configuration file,
+# see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
+#
+
+config GMOCK
+	bool "gmock"
```
Comment (pillip8282) [inline external/gmock/Kconfig]: Are there any modules which is dependent to gmock?

Comment (HONGCHAEHEE) [reply external/gmock/Kconfig]: HAVE_CXX should be enabled.

Comment (pillip8282) [reply external/gmock/Kconfig]: Then I think that adding dependency is better.
'depends on'

**Comment by sunghan-chang** [comment]
@HONGCHAEHEE You should not make "Merge branch 'master' into gmock".
Let's revert it and rebase yours.

**Pull request merged**